DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,189 @@
+  inline float getXResidual() const { return residual_.x(); }
+  inline float getYResidual() const { return residual_.y(); }
+        
+  inline float getXPull() const { return pull_.x(); }
+  inline float getYPull() const { return pull_.y(); }
+        
+  inline float getXPullNormalization() const { return residual_.x() / pull_.x(); }
+  inline float getYPullNormalization() const { return residual_.y() / pull_.y(); }
+
+  inline void setIsUsedForFit(bool usedForFit) {
+    if(usedForFit) isRealHit_ = true; 
+    isUsedForFit_ = usedForFit; 
+  }
+  inline bool getIsUsedForFit() const { return isUsedForFit_; }
+
+  inline void setIsRealHit(bool realHit) { 
+    if(!realHit) isUsedForFit_ = false;
+    isRealHit_ = realHit; 
+  }
+  inline bool getIsRealHit() const { return isRealHit_; }
+    
+private:
+  math::GlobalPoint space_point_on_det_ ;  ///< mm
+  LocalPoint residual_;  ///< mm
+  LocalPoint pull_    ;  ///< normalised residual
+  bool isUsedForFit_;
+  bool isRealHit_;
+};
+
+
+class CTPPSPixelLocalTrack: public CTPPSPixelFittedRecHit
+{
+
+  public:
+    ///< parameter vector size
+    static constexpr int dimension = 4;
+    typedef math::Error<dimension>::type CovarianceMatrix;
+    typedef math::Vector<dimension>::type ParameterVector;
+
+    ///< covariance matrix size
+    static constexpr int covarianceSize = dimension * dimension;
+
+    CTPPSPixelLocalTrack() : z0_(0), chiSquared_(0), valid_(false), numberOfPointUsedForFit_(0)
+    {
+    }
+
+    CTPPSPixelLocalTrack(float z0, const ParameterVector & track_params_vector,
+      const CovarianceMatrix &par_covariance_matrix, float chiSquared);
+
+    ~CTPPSPixelLocalTrack() override {}
+
+    inline const edm::DetSetVector<CTPPSPixelFittedRecHit>& getHits() const { return track_hits_vector_; }
+    inline void addHit(unsigned int detId, const CTPPSPixelFittedRecHit &hit)
+    {
+      track_hits_vector_.find_or_insert(detId).push_back(hit);
+      if(hit.getIsUsedForFit()) ++numberOfPointUsedForFit_;
+    }
+
+    inline float getX0() const { return track_params_vector_[0]; }
+    inline float getX0Sigma() const { return sqrt(covarianceMatrixElement(0, 0)); }
+    inline float getX0Variance() const { return covarianceMatrixElement(0, 0); }
+
+    inline float getY0() const { return track_params_vector_[1]; }
+    inline float getY0Sigma() const { return sqrt(covarianceMatrixElement(1, 1)); }
+    inline float getY0Variance() const { return covarianceMatrixElement(1, 1); }
+
+    inline float getZ0() const { return z0_; }
+    inline void setZ0(float z0) { z0_ = z0; }
+
+    inline float getTx() const { return track_params_vector_[2]; }
+    inline float getTxSigma() const { return sqrt(covarianceMatrixElement(2, 2)); }
+
+    inline float getTy() const { return track_params_vector_[3]; }
+    inline float getTySigma() const { return sqrt(covarianceMatrixElement(3, 3)); }
```

Comment:
about the "enum" comment here: the non-class enum should not need an explicit cast

Reply:
it looks like the use of enums was not considered useful

Reply:
Sorry, somehow it slipped out of the "pushes". Done now.